### PR TITLE
Bubble sort

### PR DESCRIPTION
**Pull request by NataJenkins**
Bubble sort algoritm implementation using ruby

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  i = 0
+  length = array.length - 2
+  while i <= length
+    if array[i] > array[i + 1]
+      array[i], array[i + 1] = array[i + 1], array[i]
+      i = -1
+    end
+    i += 1
+  end
+  print array
```
Comment (tirthajyoti-ghosh) [inline bubble_sort.rb]: Please remove the `print` statement and instead return ` array`. Your method must return the sorted array.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  i = 0
+  length = array.length - 2
+  while i <= length
+    if yield(array[i], array[i + 1]).positive?
+      array[i], array[i + 1] = array[i + 1], array[i]
+      i = -1
+    end
+    i += 1
+  end
+  print array
```
Comment (tirthajyoti-ghosh) [inline bubble_sort.rb]: Please remove the `print` statement and instead return `array`. Your method must return the sorted array.

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+end
+
+# first example for number
+bubble_sort_by([9, 7, 8, 4, 2, 1, 3, 5, 6])
+
+# Yield
+def bubble_sort(array)
```
Comment (tirthajyoti-ghosh) [inline bubble_sort.rb]: This method should be named `bubble_sort_by` as this method is responsible for handling `yield`.

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+def bubble_sort_by(array)
```
Comment (tirthajyoti-ghosh) [inline bubble_sort.rb]: This method should be named `bubble_sort` as this method is **not** handling `yield`.

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## What a remarkable project you have built 💯! This is impressive 🤩👌🧨!

- You have nailed the logic. The methods are working fine.

### But in order to get it approved, we need to work on some changes. Let's see...

- You have got the method names mixed up. The `bubble_sort` method should be `bubble_sort_by` and vice-versa.

- Currently, the methods are not returning anything because there is a `print` statement instead of a return statement. Remove them and return the sorted array instead.

_I have left some comments regarding the above points. Be sure to check them out._

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Review by kubilaycaglayan — APPROVED** [review]
Hello Team :wave: 
Status: Project Approved :green_circle: 

Great job on making the changes :+1:  :clap: 
:bulb: [OPTIONAL] Please fix the typo in the repository name :blush: 

With all my respect. :pray: 
> Kubilay Caglayan | Technical Support Engineer

**Pull request merged**